lib/AST/LocalizationFormat.cpp

```diff
@@ -101,12 +105,15 @@ readYAML(llvm::yaml::IO &io, T &Seq, bool, Context &Ctx) {
       DiagnosticNode current;
       yamlize(io, current, true, Ctx);
       io.postflightElement(SaveInfo);
-      // YAML file isn't guaranteed to have diagnostics in order of their
-      // declaration in `.def` files, to accommodate that we need to leave
-      // holes in diagnostic array for diagnostics which haven't yet been
-      // localized and for the ones that have `DiagnosticNode::id`
-      // indicates their position.
-      Seq[static_cast<unsigned>(current.id)] = std::move(current.msg);
+
+      if (current.id != LocalDiagID::NumDiags) {
```

Comment:
Please add a comment here specifying why this is needed and where would `id` get assigned to `NumDiags`

Reply:
What about "A diagnostic ID might be present in YAML and not in `.def` file, if that's the case `ScalarEnumerationTraits` will assign the diagnostic ID to `LocalDiagID::NumDiags`. Since the diagnostic ID it's not available in `.def` it shouldn't be saved in the diagnostics array."

Reply:
Sounds good!